corehq/apps/domain/decorators.py

```diff
@@ -178,6 +179,16 @@ def login_or_basic_ex(allow_cc_users=False):
 login_or_basic = login_or_basic_ex()
 
 
+def login_or_digest_or_basic(fn):
+    @wraps(fn)
+    def _inner(request, *args, **kwargs):
+        return {
+            'basic': login_or_basic_ex(allow_cc_users=True),
+            'digest': login_or_digest_ex(allow_cc_users=True),
+        }[determine_authtype(request)](fn)(request, *args, **kwargs)
```

Comment:
good point about ```authtype``` - forgot to deal with that part of the receiver logic.

do you have a recommendation for an alternative module?